web_api/web_api.py

```diff
@@ -0,0 +1,89 @@
+import os
+import datetime
+import uuid
+import json
+from typing import Dict, Union
+
+from flask import Flask, request, jsonify
+
+app = Flask(__name__)
+UPLOAD_FOLDER = r'C:\Users\mybj2\Desktop\School\Excellenteam\python\final-project-bezJac\uploads'
+OUTPUT_FOLDER = r'C:\Users\mybj2\Desktop\School\Excellenteam\python\final-project-bezJac\outputs'
+app.config['UPLOAD_FOLDER'] = UPLOAD_FOLDER
+app.config['OUTPUT_FOLDER'] = OUTPUT_FOLDER
+
+
+@app.route('/upload', methods=['POST'])
+def upload() -> Dict[str, str]:
+    """
+    Uploads a file to the server.
+    Returns:
+        dict: A JSON response containing the unique identifier (uid) of the uploaded file.
+    """
+    file = request.files['file']
+    if file:
+        uid = str(uuid.uuid4())
+
+        original_filename = file.filename
+        file_name, file_ext = os.path.splitext(original_filename)
+
+        timestamp = datetime.datetime.now().strftime("%Y%m%d%H%M%S")
+        new_filename = f"{uid}_{timestamp}_{file_name}_{file_ext}"
+
+        file.save(os.path.join(app.config['UPLOAD_FOLDER'], new_filename))
+
+        return jsonify({'uid': uid})
+    else:
+        return jsonify({'error': 'No file provided'}), 400
+
+
+@app.route('/status/<string:uid>', methods=['GET'])
+def status(uid: str) -> Union[Dict[str, Union[str, int, Dict[str, Union[str, int, None]]]], Dict[str, str]]:
+    """
+    Retrieves the processing status and explanation for a given file UID.
+    Args:
+        uid (str): The unique identifier of the file.
+    Returns:
+        dict: A JSON response containing the processing status, filename, timestamp, and explanation (if available).
+    """
+    uploads_path = app.config['UPLOAD_FOLDER']
+    output_path = app.config['OUTPUT_FOLDER']
+    file_exists = False
+    file_path = None
+
+    for root, dirs, files in os.walk(uploads_path):
+        for file in files:
+            if uid in file:
+                file_exists = True
+                file_path = os.path.join(root, file)
+                break
+
+    if not file_exists:
+        return jsonify({'status': 'not found'}), 404
+
+    filename = os.path.splitext(os.path.basename(file_path))[0]
+
+    timestamp = os.path.basename(file_path).split('_', 1)[1].split('_', 1)[0]
```

Comment:
Assign constants to descriptive variables to enhance readability and simplify future changes ^__^ . Directly using them can lead to confusion and maintenance challenges.